src/coprocessor/dag/batch/executors/table_scan_executor.rs

```diff
@@ -146,32 +149,26 @@ impl ScanExecutorImpl for TableScanExecutorImpl {
         let columns_len = self.schema.len();
         let mut columns = Vec::with_capacity(columns_len);
 
-        if let Some(handle_index) = self.handle_index {
-            // PK is specified in schema. PK column should be decoded and the rest is in raw format
-            // like this:
-            // non-pk non-pk non-pk pk non-pk non-pk non-pk
-            //                      ^handle_index = 3
-            //                                             ^columns_len = 7
+        let mut last_index = 0usize;
 
-            // Columns before `handle_index` (if any) should be raw.
-            for _ in 0..handle_index {
+        for handle_index in &self.handle_indices {
```

Comment:
We need to keep the order of the built column vec the same as columns given. 

Since multiple handle_indices is a very rare case, I prefer to:

```
// Fast path: self.handle_indices.len() == 1
keep the same code as current
// Fast path: self.handle_indices.len() == 0
keep the same code as current
// Slow path: self.handle_indices.len() > 1
for i in 0..self.schema {
    if i in self.handle_indices {
        push decoded column
    } else {
        push encoded column
    }
}
```

Reply:
Emmmm.. I think I kept the order. For each handle index, the loop fills non-pk columns before it and then pushes the pk column. Then the second loop pushes remaining columns after the last pk column.

Reply:
Oh. I got it! Nice solution.

Reply:
Seems I need to try to add more comments to avoid confusion